core/ast2100/ast2100.js

```diff
@@ -0,0 +1,516 @@
+"use strict";
+/*global DCTSIZE, DCTSIZE2, JpegHuffmanTable, BitStream, AST2100IDCT, AAN_IDCT_SCALING_FACTORS, ZIGZAG_ORDER */
+/*global ATEN_QT_LUMA, ATEN_QT_CHROMA, TABLE_CLASS_AC, TABLE_CLASS_DC */
+/*global BITS_AC_LUMA, BITS_AC_CHROMA, BITS_DC_LUMA, BITS_DC_CHROMA, HUFFVAL_AC_LUMA, HUFFVAL_AC_CHROMA, HUFFVAL_DC_LUMA, HUFFVAL_DC_CHROMA */
+/*global inRangeIncl, isEmpty */
+/*global fmt_u8a, fmt_u8, fmt_u16 */
+
+/*
+(c) Copyright 2015-2017 Kevin Kelley <kelleyk@kelleyk.net>.
+
+This implementation is the product of clean-room reverse engineering (that is, I
+am not and have never been subject to enondisclosure agreements, nor have I had
+access to proprietary information, related to the subject matter of this
+project).
+*/
+
+var verboseDebug = false;
+var verboseMcuCount = false;
+var traceUpdates = false;
+var verboseStats = false;
+var verboseVideoSettings = true;
+
+
+// N.B.(kelleyk): This is not properly part of the AST2100 decoder, but it's
+// very related.  This function is just a little helper to make sending these
+// messages easier.  Change the verboseVideoSettings variable above to get
+// console output showing you when the server changes what it is sending
+// (i.e. in response to this message).
+var atenChangeVideoSettings = function (lumaQt, chromaQt, subsamplingMode) {
+    // pick correct object to get the rfb websock from: in UI if enabled, in global rfb if using vnc_auto
+    var Xusock = typeof UI !== "undefined" ? UI.rfb._sock : rfb._sock;
+    RFB.messages.atenChangeVideoSettings(sock, lumaQt, chromaQt, subsamplingMode);
+};
```

Comment:
This seems unnecessary and possibly even dangerous. Encapsulate things properly in the RFB object.